internalsdk/ios/ios.go

```diff
@@ -192,10 +194,26 @@ func (c *iosClient) start() (ClientWriter, error) {
 	if err != nil {
 		return nil, err
 	}
-	dialer, err := bandit.New(dialers)
+	tracker := stats.NewTracker()
+	dialer, err := bandit.NewWithStats(dialers, tracker)
```

Comment:
@atavism I have used bandit.NewWithStats still we are not getting stats, Any Idea?

Reply:
Most likely because `GetInstance()` always returns null so we never update the session model with the new stats

Reply:
But How come tracker.AddListener is not getting called? it should at least print `log.Debugf("Received stats: %v", st)`

Reply:
How do you know `tracker.AddListener` isn't getting called? I thought we didn't see any logging when we call Go code from the PacketTunnelProvider class

Reply:
@atavism In Logs I see stats getting called, You can see the process if you attach the tunnel process on Xcode.

```
May 23 09:39:05.166 - 0m56s DEBUG bandit: bandit.go:117 Dialer jigglypuff-ac370208-aad9-4131-ab8b-d7f88211c750 dialed in 1.377574375 seconds
May 23 09:39:05.167 - 0m56s DEBUG ios: session_model.go:124 Received stats: {Frankfurt Germany DE 0 0 false false false false false connecting []}
May 23 09:39:05.167 - 0m56s DEBUG ios: session_model.go:124 Received stats: {Frankfurt Germany DE 0 0 false true false false false connected []}
```

Reply:
> You can see the process if you attach the tunnel process on Xcode.

Oh, interesting! I had no idea that was possible. Thanks for the tip